### ISSUE TITLE
assertionChallenge recommendations

### ISSUE THREAD

**Issue by apowers313** [w3c/webauthn#85]
Should we include some recommendations on the size / format of the assertion challenge? I had arbitrarily selected 256 random bytes in my server implementation, but that seems like a burden for NFC / BLE authenticators -- something that I hadn't thought about until now.

I just figure that a hint or two might prevent people who are unfamiliar with the system as a whole from creating unintentional problems.

**Comment by bifurcation** [comment]
[TPAC-2016] This should be covered by some text in the security considerations; no normative requirements.

**Comment by leshi** [comment]
The challenge will get put into the client data, which is then hashed down to a fixed-length (short) byte string. So no technical burden if the assertionChallenge is long

**Comment by AngeloKai** [comment]
Cleaning up the issues list... Assigning this to @leshi and @jcjones because they two are last seen on the thread. Based on @leshi's response, I think we can close this issue?

**Comment by equalsJeffH** [comment]
see also #404

**Comment by nadalin** [comment]
@equalsJeffH will verify and close if OK

**Comment by herrjemand** [comment]
@equalsJeffH @apowers313 I think we need to add requirements for the min length. Currently I can't see that it says anywhere anything about it.

**Comment by emlun** [comment]
There is some mention of required randomness in [§13.1. Cryptographic Challenges][sec-cons], but I agree a recommended minimum length would be good.

Should we perhaps also specify that the browser MAY, SHOULD or MUST refuse a request if the challenge is shorter than the specified minimum? Thoughts on that?

[sec-cons]: https://w3c.github.io/webauthn/#cryptographic-challenges

**Comment by herrjemand** [comment]
@emlun "Server MUST generate challenge minimum 16bytes long, and maximum of 64bytes"

**Comment by emlun** [comment]
I don't agree with a maximum length on the challenge, since only a hash of it will be sent to the authenticator, but I'll draft a proposal defining a minimum length.

**Comment by arianvp** [comment]
I was curious whether we could reconsider a maximum length on the challenge to the standard.   Firefox, Chromium and Edge all have a DoS vector when the challenge is large enough https://bugs.chromium.org/p/chromium/issues/detail?id=1048002
